File: Python/assignment4_caller.py
#  ============================================================
#  This file is part of the CS372_Code_Grader project.
#
#  Author: Omid Jafari - omidjafari.com
#  Copyright (c) 2022
#
#  For the full copyright and license information, please view
#  the LICENSE file that was distributed with this source code.
#  ============================================================

import sys
from utils import str_to_int_list
import importlib.util
from pathlib import Path


def question1_caller(input1: str, input2: int):
    output = question1(input1, input2)
    print(f"Question1: {output}")


def question2_caller(input1: str):
    output = question2(input1)
    print(f"Question2: {output}")


def question3_caller(input1: list[int], input2: list[int]):
    output = question3(input1, input2)
    print(f"Question3: {output}")


def main():
    student_code_path = Path(sys.argv[1])
    spec = importlib.util.spec_from_file_location("py_code", student_code_path)
    foo = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(foo)
    globals().update(vars(foo))

    if sys.argv[2] == "question1":
        question1_caller(sys.argv[3], int(sys.argv[4]))

    elif sys.argv[2] == "question2":
        question2_caller(sys.argv[3])

    elif sys.argv[2] == "question3":
        question3_caller(str_to_int_list(sys.argv[3]), str_to_int_list(sys.argv[4]))

    else:
        raise Exception("Question number not given.")


if __name__ == "__main__":
    main()

File: Java/src/assignment4_caller.java
//  ============================================================
//  This file is part of the CS372_Code_Grader project.
//
//  Author: Omid Jafari - omidjafari.com
//  Copyright (c) 2022
//
//  For the full copyright and license information, please view
//  the LICENSE file that was distributed with this source code.
//  ============================================================

import static assignment4.assignment4.*;

public class assignment4_caller {
    private static void question1_caller(String input1, int input2) {
        int output = question1(input1, input2);
        System.out.println("Question1: " + output);
    }

    private static void question2_caller(String input1) {
        String output = question2(input1);
        System.out.println("Question2: " + output);
    }

    private static void question3_caller(int[] input1, int[] input2) {
        int output = question3(input1, input2);
        System.out.println("Question3: " + output);
    }

    public static void main(String[] args) {
        switch (args[0]) {
            case "question1":
            	question1_caller(args[1], Integer.parseInt(args[2]));
            	break;
            case "question2":
            	question2_caller(args[1]);
            	break;
            case "question3":
            	question3_caller(utils.str_to_int_array(args[1]), utils.str_to_int_array(args[2]));
            	break;
            default:
            	throw new RuntimeException("Question number not given.");
        }
    }
}